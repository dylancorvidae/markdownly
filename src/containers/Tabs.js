import React from 'react';
import PropTypes from 'prop-types';
import TabDeck from '../components/tab/TabDeck';
import { connect } from 'react-redux';
import { getSearchTerm, getMatchTitlesByTitle } from '../selectors/documentSelectors';
import { addTab, deleteTab, addSearch, getMatchTitle } from '../actions/documentActions';
import Filter from '../components/filter/Filter';

const Tabs = ({ handleClick, titles, handleDelete, handleChange, handleSubmit, searchTerm }) => {
  return (
    <>
      <TabDeck handleDelete={handleDelete} titles={titles} />
      <button onClick={() => handleClick(titles.length)}>ADD</button>
      <Filter handleSubmit={handleSubmit} handleChange={handleChange} searchTerm={searchTerm} />
    </>
  );
};

const mapStateToProps = state => ({
  titles: getMatchTitlesByTitle(state),
  searchTerm: getSearchTerm(state)
});

const mapDispatchToProps = dispatch => ({
  handleClick(tabNumber) {
    dispatch(addTab(tabNumber));
  },
  handleDelete(title) {
    dispatch(deleteTab(title));
  },
  handleSubmit(searchTerm) {
    event.preventDefault();
    dispatch(getMatchTitle(searchTerm));
  },
  handleChange({ target }) {
    event.preventDefault();
    dispatch(addSearch(target.value));
  }
});

Tabs.propTypes = {
  titles: PropTypes.array.isRequired,
  handleClick: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  searchTerm: PropTypes.string
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Tabs);


