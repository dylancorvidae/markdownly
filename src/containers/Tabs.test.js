import React from 'react';
import { shallow } from 'enzyme';
import Tabs from './Tabs';
import store from '../store';
import { Provider } from 'react-redux';

describe('Tabs component', () => {
  it('renders Tabs', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Tabs />);
      </Provider>);
    expect(wrapper).toMatchSnapshot();
  });
}); 
