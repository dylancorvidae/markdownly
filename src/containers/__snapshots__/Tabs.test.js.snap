// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders Tabs 1`] = `
<ContextProvider
  value={null}
>
  <Tabs
    handleChange={[Function]}
    handleClick={[Function]}
    handleDelete={[Function]}
    handleSubmit={[Function]}
    searchTerm=""
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    titles={
      Array [
        "default",
      ]
    }
  />
</ContextProvider>
`;
